snuba/manual_jobs/update_migration_status.py

```diff
@@ -0,0 +1,96 @@
+
+from snuba.clusters.cluster import ClickhouseClientSettings, get_cluster
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.manual_jobs import Job, JobSpec, logger
+from snuba.migrations.runner import DIST_TABLE_NAME, LOCAL_TABLE_NAME
+
+
+class UpdateMigrationStatus(Job):
+    """
+    UpdateMigrationStatus is a manual job for correcting the status of an
+    entry in the migrations_[local|dist] table, to be run via the snuba jobs CLI
+    or via snuba admin (this should exclusively be instantiated from the manual
+    jobs runner, not imported by other packages).
+
+    It requires the following parameters:
+        - group - migration group that the entry belongs to
+        - migration_id - specific ID for the entry
+        - old_status - the current entry's status in the table (to make sure the user understands what they're
+          updating from)
+        - new_status - the status that the entry should contain after the update is run.
+
+    It uses an INSERT rather than an ALTER TABLE ... UPDATE in order to keep record-keeping
+    in the migrations table sensible.
+    """
+
+    def __init__(
+        self,
+        job_spec: JobSpec,
+        dry_run: bool,
+    ):
+        self.__validate_job_params(job_spec.params)
+        super().__init__(job_spec, dry_run)
+
+    def __validate_job_params(self, params: Optional[Mapping[Any, Any]]) -> None:
+        assert (
+            params is not None
+        ), "group, migration_id, old_status, new_status parameters required"
+        assert params["group"], "group required"
+        assert params["migration_id"], "migration_id required"
+        assert params["new_status"], "new_status required"
+        assert params["old_status"], "old_status required"
+        self._group = params["group"]
+        self._migration_id = params["migration_id"]
+        self._new_status = params["new_status"]
+        self._old_status = params["old_status"]
+        assert isinstance(self._group, str)
+        assert isinstance(self._migration_id, str)
+        assert isinstance(self._new_status, str)
+        assert isinstance(self._old_status, str)
+
+    def _select_query(self, table_name: str) -> str:
+        return f"SELECT status, version FROM {table_name} FINAL WHERE group = %(group)s AND migration_id = %(migration_id)s;"
+
+    def _insert_query(self, table_name: str) -> str:
+        return f"INSERT INTO {table_name} FORMAT JSONEachRow"
+
+    def execute(self) -> None:
+        migrations_cluster = get_cluster(StorageSetKey.MIGRATIONS)
+        table_name = (
+            LOCAL_TABLE_NAME if migrations_cluster.is_single_node() else DIST_TABLE_NAME
+        )
+        connection = migrations_cluster.get_query_connection(
+            ClickhouseClientSettings.MIGRATE
+        )
+
+        existing_row = connection.execute(
+            self._select_query(table_name),
+            {"migration_id": self._migration_id, "group": self._group},
+        )
+        assert (
```

Comment:
Question: The output of the negative case in the PR description dumps a stacktrace on terminal. Should we raise exception with error messages instead of asserting?

Reply:
this assert will raise an exception with the message 

"cannot find row with group {self._group}, ID {self._migration_id} in table {table_name}: no update to be performed"

here's an example from my local console:
```
➜  snuba git:(job-runner/create-migration-overwrite-job) ✗ snuba jobs run --job_type=UpdateMigrationStatus --job_id=test3 migration_id=0020_transactions_add_codecs old_status=completed new_status=not_started group=transactions
2024-10-03 10:09:24,122 Initializing Snuba...
2024-10-03 10:09:25,837 Snuba initialization took 1.716169028004515s
2024-10-03 10:09:25,887 Job failed
Traceback (most recent call last):
  File "/Users/oliver/workplace/snuba/snuba/manual_jobs/runner.py", line 145, in run_job
    job_to_run.execute()
  File "/Users/oliver/workplace/snuba/snuba/manual_jobs/update_migration_status.py", line 78, in execute
    status == self._old_status
AssertionError: actual status not_started does not match expected completed, aborting
```